modules/javafx.media/src/main/native/jfxmedia/platform/osx/avf/AVFAudioEqualizer.cpp

```diff
@@ -371,36 +359,79 @@ void AVFAudioEqualizer::ResetBandParameters() {
         }
 
         band->SetFilterType(type);
-        band->SetChannelCount(GetChannelCount());
+        band->SetChannelCount(GetChannels());
         band->RecalculateParams();
         iter++; // here due to NULL ptr protection, otherwise we double increment
     }
+
+    // Clear temp buffers
+    if (mEQBufferA != NULL) {
+        memset(mEQBufferA, 0, mEQBufferSize * sizeof(double));
+    }
+
+    if (mEQBufferB != NULL) {
+        memset(mEQBufferB, 0, mEQBufferSize * sizeof(double));
+    }
 }
 
-void AVFAudioEqualizer::SetAudioUnit(AUEffectBase *unit) {
-    this->AVFKernelProcessor::SetAudioUnit(unit);
-    ResetBandParameters();
+void AVFAudioEqualizer::SetSampleRate(UInt32 rate) {
+    mSampleRate = rate;
+}
+
+void AVFAudioEqualizer::SetChannels(UInt32 count) {
+    mChannels = count;
+}
+
+UInt32 AVFAudioEqualizer::GetSampleRate() {
+    return mSampleRate;
+}
+
+UInt32 AVFAudioEqualizer::GetChannels() {
+    return mChannels;
+}
+
+bool AVFAudioEqualizer::ProcessBufferLists(const AudioBufferList & buffer,
+                                               UInt32 inFramesToProcess) {
+    for (UInt32 i = 0; i < buffer.mNumberBuffers; i++) {
+        RunFilter((const Float32 *) buffer.mBuffers[i].mData,
+                  (Float32 *) buffer.mBuffers[i].mData,
+                  inFramesToProcess,
+                  i);
+    }
+
+    return true;
 }
 
 void AVFAudioEqualizer::RunFilter(const Float32 *inSourceP,
-                                  Float32 *inDestP,
-                                  UInt32 inFramesToProcess,
-                                  UInt32 channel) {
+        Float32 *inDestP,
+        UInt32 inFramesToProcess,
+        UInt32 channel) {
     if (mEnabled && !mEQBands.empty()) {
         if (inFramesToProcess + 2 > mEQBufferSize) {
             mEQBufferSize = inFramesToProcess + 2;
-            mEQBufferA.free();
-            mEQBufferA.alloc(mEQBufferSize);
-            mEQBufferB.free();
-            mEQBufferB.alloc(mEQBufferSize);
+            if (mEQBufferA != NULL) {
+                free(mEQBufferA);
+                mEQBufferA = NULL;
+            }
+            if (mEQBufferB != NULL) {
+                free(mEQBufferB);
+                mEQBufferB = NULL;
+            }
+
+            mEQBufferA = (double*)calloc(mEQBufferSize, sizeof(double));
+            mEQBufferB = (double*)calloc(mEQBufferSize, sizeof(double));
```

Comment:
This is allocating 8 times (i.e., `sizeof(double)`) as much memory as before. I presume this is intentional?

Reply:
No, it is allocating same amount of memory as before, unless I am missing something.
We used to have CAAutoFree<double> mEQBufferA and
CAAutoFree.alloc(size_t numItems) method allocates (numItems * sizeof(T)) bytes. See deleted CAAutoDisposer.h for CAAutoFree class implementation.

Reply:
Yes, you're right. I missed that the previous `alloc` call was a method call on a CAAutoFree object, which does exactly what you said above. The new code is equivalent.